src/Application/Services/ChatHistoryService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Application.Contracts;
+using Domain;
+using Domain.Entities;
+using Newtonsoft.Json;
+using SharedClasses;
+
+namespace Application.Services;
+
+public class ChatHistoryService
+{
+    private readonly IRulesContext _context;
+    private readonly ICurrentUserService _currentUserService;
+    private readonly SemanticKernelService _semanticKernelService;
+
+    private string Email
+    {
+        get
+        {
+            var email = _currentUserService.GetEmail();
+            if (email == null)
+                throw new ArgumentException("No email found for user!");
+            
+            return email;
+        }
+    }
+
+    public ChatHistoryService(
+        IRulesContext context, 
+        ICurrentUserService currentUserService,
+        SemanticKernelService semanticKernelService)
+    {
+        _context = context;
+        _currentUserService = currentUserService;
+        _semanticKernelService = semanticKernelService;
+    }
+
+    public ConversationHistoryModel? GetConversation(int id)
+    {
+        return _context.ConversationHistories
+            .FirstOrDefault(s => s.Id == id && s.User == Email);
+    }
+
+    public IEnumerable<ChatHistoryDetail> GetConversations()
+    {
+        return _context.ConversationHistories
+            .Where(s => s.User == Email)
+            .OrderByDescending(s => s.Date)
+            .Select(s =>
+                new ChatHistoryDetail
+                {
+                    Id = s.Id,
+                    ConversationTitle = s.ConversationTitle
+                });
+    }
+
+    public async Task AddConversation(string conversation, string firstMessage)
+    {
+        ValidateConversation(conversation);
+
+        var title = string.IsNullOrWhiteSpace(firstMessage) 
+            ? "New conversation"
+            : await GetConversationTitle(firstMessage);
+        
+        if (string.IsNullOrWhiteSpace(title))
+            title = "New conversation";
+
+        _context.ConversationHistories.Add(
+            new ConversationHistoryModel
+            {
+                Date = DateTimeOffset.Now.ToUniversalTime(),
+                ConversationTitle = title,
+                Conversation = conversation,
+                User = Email,
+                SchemaVersion = 1,
```

Comment:
Should SchemaVersion be stored in a constant somewhere?

Reply:
As per my conversation with @Hona, serializing the class name is probably not necessary as this will only ever be one class.
I've moved the ScemaVersion to a constant in the same file with a note to extract it out if it ends up being used.